lib/rmt/cli/mirror.rb

```diff
@@ -70,34 +68,51 @@ def product(*targets)
         products = Product.get_by_target!(target)
         raise RMT::CLI::Error.new(_('Product for target %{target} not found') % { target: target }) if products.empty?
         products.each do |product|
-          product_repos = product.repositories.where(mirroring_enabled: true)
+          product_repos = product.repositories.only_mirroring_enabled
           raise RMT::CLI::Error.new(_('Product %{target} has no repositories enabled') % { target: target }) if product_repos.empty?
           repos += product_repos.to_a
         end
       rescue ActiveRecord::RecordNotFound
         raise RMT::CLI::Error.new(_('Product with ID %{target} not found') % { target: target })
       end
 
-      repos.each do |repo|
-        mirror_repo!(mirror, repo)
-      rescue RMT::Mirror::Exception => e
-        logger.warn e.to_s
-      end
+      _, repo_errors = mirror_repos!(mirror, repos)
+      handle_errors!(repo_errors)
     end
   end
 
   protected
 
-  def mirror_repo!(mirror, repo)
-    mirror.mirror(
-      repository_url: repo.external_url,
-      local_path: Repository.make_local_path(repo.external_url),
-      auth_token: repo.auth_token,
-      repo_name: repo.name
-    )
+  def mirror_repos!(mirror, repos)
+    mirrored_repo_ids = []
+    errors = []
+
+    repos.each do |repo|
+      mirror.mirror(
+        repository_url: repo.external_url,
+        local_path: Repository.make_local_path(repo.external_url),
+        auth_token: repo.auth_token,
+        repo_name: repo.name
+      )
+      repo.refresh_timestamp!
+    rescue RMT::Mirror::Exception => e
+      errors << _("Repository '%{repo_name}' (%{repo_id}): %{error_message}") % {
+        repo_id: repo.id, repo_name: repo.name, error_message: e.message
+      }
+    ensure
+      mirrored_repo_ids << repo.id
+    end
 
-    repo.refresh_timestamp!
+    [mirrored_repo_ids, errors]
   end
 
+  def handle_errors!(errors)
+    return if errors.empty?
 
+    raise RMT::CLI::Error.new(
+      _('The following errors ocurred while mirroring:%{errors_list}') % {
+        errors_list: "\n" + errors.join("\n")
```

Comment:
I think that we probably want one extra \n at the start. Also, we could consider colors like

```
def handle_errors!(errors)
    return if errors.empty?

    raise RMT::CLI::Error.new(
      _('The following errors ocurred while mirroring:%{errors_list}') % {
        errors_list: "\n\e[31m" + errors.join("\n") + "\e[0m"
      }
    )
  end
```

Also, consider having a "mirroring complete" message.